File: 861-flipping-an-image/flipping-an-image.java
class Solution {
    public int[][] flipAndInvertImage(int[][] image) {
        for (int[] row: image){
            int left = 0, right = row.length -1;
            while (left <= right){
                int temp = row[left] ^ 1;
                row[left] = row[right] ^ 1;
                row[right] = temp;
                left += 1;
                right -= 1;
            }
        }
        return image;
    }
}

File: 861-flipping-an-image/flipping-an-image.py
class Solution:
    def flipAndInvertImage(self, image: List[List[int]]) -> List[List[int]]:
        newImage = []
        n = len(image)
        for i in range(n):
            temp = image[i][::-1]
            for j in range(n):
                temp[j] = ~temp[j] & 1
            newImage.append(temp)
        return newImage


